libs/optimizer/src/lib/optimizer.ts

```diff
@@ -231,7 +231,7 @@ const circuitMapping = {
 };
 
 function toCircuitType(code: string): CircuitType {
-  const type = circuitMapping[code];
+  const type = circuitMapping[code as 'od' | 'tri' | 'fai' | 'oar'];
```

Comment:
-1 on this.
If there such be such type it should be on the parameter.
But I don't think we need a cast here.
How do you get an error?

Reply:
error occurred when running `npm run dev`